developer-docs-site/docs/guides/move-guides/move-scripts.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "Move Scripts"
+slug: "move-scripts"
+---
+
+# Move Scripts
+
+You can use Move scripts to conduct arbitrary tasks on chain. 
+
+## Example use case
+
+The following example calls functions on the [aptos_coin.move](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/aptos_coin.move) module to confirm the balance of the destination account is less than `desired_balance`, and if so, tops it up to `desired_balance`.
+
+```move
+script {
+    use std::signer;
+    use aptos_framework::aptos_account;
+    use aptos_framework::aptos_coin;
+    use aptos_framework::coin;
+
+    fun main(src: &signer, dest: address, desired_balance: u64) {
+        let src_addr = signer::address_of(src);
+
+        let balance = coin::balance<aptos_coin::AptosCoin>(src_addr);
+        if (balance < desired_balance) {
+            aptos_account::transfer(src, dest, desired_balance - balance);
+        };
+
+        addr::my_module::do_nothing();
+    }
+}
+```
+
+Notice also how it calls a function in a Move module defined as:
+```
+module addr::my_module {
+    public entry fun do_nothing() { }
+}
+```
+
+Now you need to determine:
+
+- Where do I put these files?
+- Do I need a `Move.toml`?
+- How do I run my script with the CLI?
+
+## Execution
+
+Let's run through how to execute a Move script with a step-by-step example using the [Aptos CLI](../../cli-tools/aptos-cli-tool/use-aptos-cli.md).
+
+Make a new directory for your work:
+```sh
+mkdir testing
+cd testing
+```
+
+Set up the Aptos CLI:
+```sh
+aptos init
+```
+The CLI will ask you which network you want to work with (e.g. `devnet`, `testnet`, `mainnet`). It will also ask you for your private key (which looks like this: `0xf1adc8d01c1a890f17efc6b08f92179e6008d43026dd56b71e7b0d9b453536be`), or it can generate a new one for you, as part of setting up your account.
+
+From here, initialize a new Move project:
+```sh
+aptos move init --name my_script
+```
+
+Now create a `top_up.move` file containing the example script above in a `sources/` subdirectory of your `testing/` directory, like so:
+```
+testing/
+   Move.toml
+   sources/
+      top_up.move
+```
+
+Now do the same thing with the Move module, leaving you with this:
```

Comment:
I find this line confusing. Do exactly what?

Also consider including the contents of the suggested `Move.toml` say from:
https://github.com/banool/move-examples/blob/main/run_script/Move.toml

Reply:
Yeah I agree this is a bit confusing. In the initial question there is a script in the first code block, and then a move module in the second code block. Starting from line 68 I'm saying you should put the code in the first block in `top_up.move`. Then here I'm saying we should put the content of the second code block in `my_module.move`. The thing is, the Q&A format is such that in the question, you're assuming that the user already has the code in those two code blocks, so this is just a rehash. If you rewrite this as a guide / tutorial (which I see we're sort of doing already with the new section titles), that flow maybe makes less sense.

Reply:
Ah, thanks Daniel. This is what I needed to know. I have tried to make all of that more clear and also include creation of the `Move.toml` file as an explicit step. Please take another look. I can return this to a more Q&A format and tried to avoid unnecessary changes. Thanks again!

Reply:
I ended up removing the Move module part following a comment from Kevin.

Reply:
OK, I still see the `Move.toml` file example I added. Please remove if irrelevant now.

Reply:
OK, resolving this convo. :-) Thanks!